source/common/upstream/load_balancer_impl.cc

```diff
@@ -1299,25 +1299,88 @@ HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector
                                                                 const HostsSource&) {
   HostSharedPtr candidate_host = nullptr;
 
+  switch (selection_method_) {
+    case envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest_SelectionMethod_FULL_SCAN:
+      candidate_host = unweightedHostPickFullScan(hosts_to_use);
+      break;
+    case envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest_SelectionMethod_N_CHOICES:
+    default:
+      candidate_host = unweightedHostPickNChoices(hosts_to_use);
+  }
+
+  return candidate_host;
+}
+
+HostSharedPtr LeastRequestLoadBalancer::unweightedHostPickFullScan(const HostVector& hosts_to_use) {
+  HostSharedPtr candidate_host = nullptr;
+
+  // "Randomly" scan the host array using a linear congruential generator with the following parameters:
+  // - The "modulus" is the number of hosts
+  // - The "multiplier" is 1
+  // - The "increment" is a non-zero value less than the modulus.
+  //   It is guaranteed non-zero because it is determined by:
+  //     random large prime number / modulus ... which never be 0
+  // - The "start value" is a random index in the host array
+  //
+  // With these parameters, the generator has a period equal to the modulus (number of hosts).
+  //
+  // The effect of this algorithm is to visit each index of the host array exactly once.
+  // The next index will always be +"increment" indices from the current index.
+  //
+  // This algorithm is not *very* random, but it is random *enough* for its purpose.
+  // Approaches with more randomness were not used because they are less performant for larger host arrays
+  // (e.g., requiring the allocation of an array the length of the host array).
+
+
+  // Because the initial value of host_scan_index will always be the last index scanned,
+  // it must be a random index.
+  unsigned long host_scan_index = random_.random() % hosts_to_use.size();
```

Comment:
nit: use uintXX_t here explicitly. Our style is to be explicit and also just to be consistent with other code this this file.

Reply:
Sounds good to me :) That's also my preference.